tests/formats/format_helpers.test.ts

```diff
@@ -317,18 +339,44 @@ describe("formatValue on number", () => {
     );
   });
 
-  test("with brackets inside the string", () => {
-    expect(() => formatValue(1234, { format: "[$[]#,##0.0", locale })).toThrow();
-    expect(() => formatValue(1234, { format: "[$]]#,##0.0", locale })).toThrow();
-    expect(() => formatValue(1234, { format: "[$[]]#,##0.0", locale })).toThrow();
-    expect(() => formatValue(1234, { format: "[$][]#,##0.0", locale })).toThrow();
+  test("Can escape strings or characters in format", () => {
+    expect(formatValue(5, { format: "0\\%", locale })).toBe("5%");
+    expect(formatValue(5, { format: "0\\€", locale })).toBe("5€");
+    expect(() => formatValue(5, { format: "0€", locale })).toThrow();
+    expect(formatValue(5, { format: "0[$€]", locale })).toBe("5€");
+    expect(formatValue(5, { format: '0"€"', locale })).toBe("5€");
+    expect(formatValue(5, { format: '0"+"', locale })).toBe("5+");
+
+    expect(() => formatValue(5, { format: '0"€', locale })).toThrow();
+    expect(() => formatValue(5, { format: "0[$", locale })).toThrow();
+    expect(() => formatValue(5, { format: "0\\", locale })).toThrow();
+  });
```

Comment:
Concerning escaped string, maybe add test for concrete cases desired functionally:
![image](https://github.com/user-attachments/assets/b9588a93-bfa6-485f-b23d-d0a5d88671d3)